src/Data/Acid/Log.hs

```diff
@@ -193,6 +194,19 @@ readEntriesFrom fLog youngestEntry = do
  where
   rangeStart (firstEntryId, _path) = firstEntryId
 
+-- Read a specific Entry or Nothing.
+readEntry :: SafeCopy object => FileLog object -> EntryId -> IO (Maybe object)
+readEntry fLog entryId = do
+  -- We're interested in the first entry of all entries: youngestEntry <= x.
+  logFiles <- findLogFiles (logIdentifier fLog)
+  let sorted = sort logFiles
+      logFile = listToMaybe $ filterLogFiles (Just entryId) (Just (entryId+1)) sorted
```

Comment:
This function is probably not safe. It will fail hard if the log file that contains the entry is currently still being written too, and it assumes that the first entry of the log file is the one we're looking for.

However this seems to be always true for checkpoints. Might still make sense to add some safeguards here, and especially document these restrictions. There might also be a better way of implementing this instead, by explicitly looking at the newest log file that is not the current one (i.e. written to) and getting the last entry from it only. Basically making this a readNewestDurableEntry.

OTOH this all seems overly complicated, why don't we just remember the id of the latest checkpoint together with the last entry id contained in it in the state?